docs/topics/native/native-c-interop.md

```diff
@@ -429,36 +396,63 @@ fun zeroMemory(buffer: COpaquePointer, size: Int) {
 }
 ```
 
-Also, the type parameter can be inferred automatically and so may be omitted
-in some cases.
+Also, the type parameter can be inferred automatically and so may be omitted in some cases.
 
 ### Object pinning
 
-Kotlin objects could be pinned, i.e. their position in memory is guaranteed to be stable
-until unpinned, and pointers to such objects inner data could be passed to the C functions. For example
-
-```kotlin
-import kotlinx.cinterop.*
-import platform.posix.*
-
-@OptIn(ExperimentalForeignApi::class)
-fun readData(fd: Int) {
-    val buffer = ByteArray(1024)
-    buffer.usePinned { pinned ->
-        while (true) {
-            val length = recv(fd, pinned.addressOf(0), buffer.size.convert(), 0).toInt()
-
-            if (length <= 0) {
-               break
-            }
-            // Now `buffer` has raw data obtained from the `recv()` call.
-        }
-    }
-}
-```
-
-Here we use service function `usePinned`, which pins an object, executes block and unpins it on normal and
-exception paths.
+Kotlin objects could be pinned, i.e. their position in memory is guaranteed to be stable until they are unpinned,
+and pointers to such objects' inner data could be passed to C functions.
+
+There's a couple of approaches you can take:
+
+* Use the [`usePinned`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlinx.cinterop/use-pinned.html) service function
+  that pins an object, executes a block, and unpins it on normal and exception paths:
+
+  ```kotlin
+  import kotlinx.cinterop.*
+  import platform.posix.*
+
+  @OptIn(ExperimentalForeignApi::class)
+  fun readData(fd: Int) {
+      val buffer = ByteArray(1024)
+      buffer.usePinned { pinned ->
+          while (true) {
+              val length = recv(fd, pinned.addressOf(0), buffer.size.convert(), 0).toInt()
+              if (length <= 0) {
+                  break
+              }
+              // Now `buffer` has raw data obtained from the `recv()` call.
+          }
+      }
+  }
+  ```
+
+  Here, `pinned` is an object of a special type `Pinned<T>`. It provides useful extensions like `addressOf`, which allows
+  getting the address of a pinned array body.
+
+* Use the [`refTo()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlinx.cinterop/ref-to.html) function that has
+  similar functionality under the hood but, in certain cases, may help you reduce boilerplate code:
+
+  ```kotlin
+  import kotlinx.cinterop.*
+  import platform.posix.*
+    
+  @OptIn(ExperimentalForeignApi::class)
+  fun readData(fd: Int) { 
+  val buffer = ByteArray(1024)
+      while (true) {
+          val length = recv(fd, buffer.refTo(0), buffer.size.convert(), 0).toInt()
+
+          if (length <= 0) {
+              break
+          }
+          // Now `buffer` has raw data obtained from the `recv()` call.
+      }
+  }
+  ```
+
+  Here, `buffer.refTo(0)` has the `CValuesRef` type that pins the array before entering the function, passes the address
```

Comment:
```suggestion
  Here, `buffer.refTo(0)` has the `CValuesRef` type that pins the array before entering the `recv` function, passes the address
```

Just for the sake of being abundantly clear (and if I didn't understand correctly which function we're entering, then readers might not, too :).